internals/daemon/daemon.go

```diff
@@ -228,14 +228,11 @@ func (c *Command) Daemon() *Daemon {
 }
 
 func (c *Command) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	st := c.d.state
-	st.Lock()
-	user, err := userFromRequest(st, r)
+	user, err := userFromRequest(nil, r) // don't pass state as this does nothing right now
 	if err != nil {
 		statusForbidden("forbidden").ServeHTTP(w, r)
 		return
 	}
-	st.Unlock()
```

Comment:
This is a drive-by for something that's not actually an issue, but seems like an accident waiting to happen -- we're not locking in the case where `userFromRequest` function returns an error. In this case doing `userFromRequest` is doing nothing, so just avoid locking altogether (can always add the lock back later as needed).

Reply:
we're not unlocking* in the case where `userFromRequest` function returns an error